components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/DashboardDataProviderAuthorizer.java

```diff
@@ -294,11 +298,14 @@ private void assembleQuery(String username, DataProviderConfigRoot dataProviderC
         if (tenantDomain != null && !tenantDomain.isEmpty()) {
             if (tenantDomain.equalsIgnoreCase(SUPER_TENANT_DOMAIN)) {
                 contextPath = NOT_LIKE_CONTEXT_PATH;
+                contextContainsCondition = STRING_NOT_CONTAIN_CONTEXT;
             } else {
                 contextPath = LIKE_CONTEXT_PATH;
+                contextContainsCondition = STRING_CONTAIN_CONTEXT;
             }
             String tenantId = getTenantId(username);
             query = query.replace(CONTEXT_CONDITION_KEY, contextPath)
```

Comment:
Consider replacing this with StringBuilder if the replace is likely to grow.

Reply:
This replace will not grow. If we use StringBuilder, we need to provide the starting point and end for each of the replace action and since it makes the code ugly, kept the String replace.